### PR TITLE
Refactor: remove `@ts-expect-error`

### PR DESCRIPTION
**Pull request by sosukesuzuki**
## Checklist

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory).
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/main/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by fisker** [comment]
https://github.com/prettier/prettier/pull/11597/files#diff-0a9f43535bdd52514d546791181ecaf988bb8215414f8baeb7efc43fb9b1896aR44

File: src/language-markdown/utils.js [hunk 1]
```diff
@@ -152,11 +151,16 @@ function splitText(text, options) {
...
 
-    function isBetween(kind1, kind2) {
+    /**
+     * @param {WordNode} lastNode
+     * @param {any} node
+     * @param {string} kind1
+     * @param {string} kind2
+     * @returns
+     */
+    function isBetween(lastNode, node, kind1, kind2) {
```
Comment (fisker) [inline src/language-markdown/utils.js:161]: I really don't know types, and even don't like types. But this is really silly to pass a local variable to the function.

File: src/language-markdown/utils.js [hunk 2]
```diff
@@ -152,11 +151,16 @@ function splitText(text, options) {
     }
     nodes.push(node);
 
-    function isBetween(kind1, kind2) {
+    /**
+     * @param {WordNode} lastNode
```
Comment (fisker) [inline src/language-markdown/utils.js:155]: Are you sure lastNode can only be `WordNode`?

**Comment by sosukesuzuki** [comment]
Thanks for the review. Anyway, I can close this PR as it looks like it will be fixed in #11597

File: src/language-markdown/utils.js (hunk 1 above)
Comment (sosukesuzuki) [reply src/language-markdown/utils.js:161]: I agree with that, but I don't like using ts-expect-error or ts-ignore more than necessary.

**Pull request closed without merging**